pkg/api/api_v2.go

```
@@ -1082,3 +1087,152 @@ func (api *v2api) userEnvPodLogsHandler(w http.ResponseWriter, r *http.Request)
+		api.rlogger(r).Logf("action not authorized by user: %v", uis.GitHubUser)
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+	v2uakds := make([]V2UserAPIKeyData, 0, len(apikeys))
+	for _, apikey := range apikeys {
+		v2uakds = append(v2uakds, V2UserAPIKeyData{
+			ID:          apikey.ID,
+			Created:     apikey.Created,
+			LastUsed:    apikey.LastUsed,
+			Permission:  apikey.PermissionLevel,
+			Description: apikey.Description,
+			User:        apikey.GitHubUser,
+		})
+	}
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(&v2uakds); err != nil {
+		api.rlogger(r).Logf("error marshaling new user token: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+	}
+}
+
+func (api *v2api) apiKeyDestroyHandler(w http.ResponseWriter, r *http.Request) {
+	uis, err := getSessionFromContext(r.Context())
+	if err != nil {
+		api.rlogger(r).Logf("session missing from context")
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	id := mux.Vars(r)["id"]
+	if id == "" {
+		api.rlogger(r).Logf("error token id required")
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	keyID, err := uuid.Parse(id)
+	if err != nil {
+		api.rlogger(r).Logf("error parsing user token: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	apikey, err := api.dl.GetAPIKeyByID(r.Context(), keyID)
+	if err != nil {
+		api.rlogger(r).Logf("error getting api key from db: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	if apikey == nil {
+		api.rlogger(r).Logf("no key found for id: %v", id)
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+	if uis.GitHubUser != apikey.GitHubUser {
+		api.rlogger(r).Logf("action not authorized by user: %v", uis.GitHubUser)
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+	err = api.dl.DeleteAPIKeyByID(context.Background(), apikey.ID)
+	if err != nil {
+		api.logger.Printf("error deleting user token: %v: %v", id, err)
```

You probably should return a 500 error here.